api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -346,9 +343,13 @@ public boolean checkRequireExpression(Requireable candidate, AppContextModel con
 					} else {
 						bindings.put(e.getKey(), e.getValue());
 					}
-				}
+			}
+
+			for (Map.Entry<String, Object> e : contextModel.entrySet()) {
```

Comment:
Do you also mind explaining the motivation for adding this?

Reply:
I think GraalJS handles datatypes like Map differently. So without this, we get errors similar to this: https://pastebin.com/A2YtRDGY .

Reply:
Isn't this duplicating some of what is already done on line 356?

Reply:
How about simply replacing it with this? https://pastebin.com/BDgMc9Hj